### ISSUE TITLE
Undefined array key listener - moving a file to a folder

### ISSUE THREAD

**Issue by stuartcusackie** [statamic/cms#7108]
### Bug description

I'm getting a red alert box when I try to move an image to a folder using the dialog. "Undefined array key listener".

![image](https://user-images.githubusercontent.com/20210934/203784051-807eccaa-e08a-4828-9369-11ea8efb06d5.png)

- The file does in fact move but references to the file in my entries are not updated - This is a bit worrying because it could break a lot of entries / pages if a client decided to move a lot of files to a folder. 
- I am using DO Spaces for this container.
- Happening on my local environment (not tested on remote).

### How to reproduce

Try moving a file to a folder using the 'Move' option in the dropdown.

### Logs

```
[2022-11-24 12:23:38] local.ERROR: Undefined array key "listener" {"userId":"d0370ca3-200c-4070-bd11-cd641ae5d8de","exception":"[object] (ErrorException(code: 0): Undefined array key \"listener\" at C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Events\\Subscriber.php:86)
[stacktrace]
#0 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Bootstrap\\HandleExceptions.php(259): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError(2, 'Undefined array...', 'C:\\\\webdev\\\\larag...', 86)
#1 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Events\\Subscriber.php(86): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->Illuminate\\Foundation\\Bootstrap\\{closure}(2, 'Undefined array...', 'C:\\\\webdev\\\\larag...', 86)
#2 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Providers\\EventServiceProvider.php(47): Statamic\\Events\\Subscriber::normalizeRegisteredListener(Object(Closure))
#3 [internal function]: Illuminate\\Events\\Dispatcher->Statamic\\Providers\\{closure}(Object(Closure), 6)
#4 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Collections\\Arr.php(795): array_filter(Array, Object(Closure), 1)
#5 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Providers\\EventServiceProvider.php(48): Illuminate\\Support\\Arr::where(Array, Object(Closure))
#6 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Macroable\\Traits\\Macroable.php(124): Illuminate\\Events\\Dispatcher->Statamic\\Providers\\{closure}('Statamic\\\\Events...', 'Statamic\\\\Git\\\\Su...')
#7 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Events\\Subscriber.php(49): Illuminate\\Events\\Dispatcher->__call('forgetListener', Array)
#8 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Events\\Subscriber.php(68): Statamic\\Events\\Subscriber::disable()
#9 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Data\\DataReferenceUpdater.php(287): Statamic\\Events\\Subscriber::withoutListeners(Object(Closure))
#10 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Data\\DataReferenceUpdater.php(66): Statamic\\Data\\DataReferenceUpdater->saveItem()
#11 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Listeners\\UpdateAssetReferences.php(99): Statamic\\Data\\DataReferenceUpdater->updateReferences('poet-patrick-ka...', 'patrick-kavanag...')
#12 [internal function]: Statamic\\Listeners\\UpdateAssetReferences->Statamic\\Listeners\\{closure}(Object(Statamic\\Entries\\Entry), 6)
#13 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Collections\\Arr.php(560): array_map(Object(Closure), Array, Array)
#14 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Collections\\Collection.php(739): Illuminate\\Support\\Arr::map(Array, Object(Closure))
#15 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Listeners\\UpdateAssetReferences.php(100): Illuminate\\Support\\Collection->map(Object(Closure))
#16 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Listeners\\UpdateAssetReferences.php(48): Statamic\\Listeners\\UpdateAssetReferences->replaceReferences(Object(Statamic\\Assets\\Asset), 'poet-patrick-ka...', 'patrick-kavanag...')
#17 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Events\\CallQueuedListener.php(107): Statamic\\Listeners\\UpdateAssetReferences->handleSaved(Object(Statamic\\Events\\AssetSaved))
#18 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Container\\BoundMethod.php(36): Illuminate\\Events\\CallQueuedListener->handle(Object(Illuminate\\Foundation\\Application))
#19 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Container\\Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#20 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Container\\BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure(Object(Closure))
#21 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Container\\BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Array, Object(Closure))
#22 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Container\\Container.php(651): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Array, Array, NULL)
#23 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Bus\\Dispatcher.php(128): Illuminate\\Container\\Container->call(Array)
#24 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(141): Illuminate\\Bus\\Dispatcher->Illuminate\\Bus\\{closure}(Object(Illuminate\\Events\\CallQueuedListener))
#25 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Events\\CallQueuedListener))
#26 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Bus\\Dispatcher.php(132): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#27 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Queue\\CallQueuedHandler.php(124): Illuminate\\Bus\\Dispatcher->dispatchNow(Object(Illuminate\\Events\\CallQueuedListener), false)
#28 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(141): Illuminate\\Queue\\CallQueuedHandler->Illuminate\\Queue\\{closure}(Object(Illuminate\\Events\\CallQueuedListener))
#29 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Events\\CallQueuedListener))
#30 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Queue\\CallQueuedHandler.php(126): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#31 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Queue\\CallQueuedHandler.php(70): Illuminate\\Queue\\CallQueuedHandler->dispatchThroughMiddleware(Object(Illuminate\\Queue\\Jobs\\SyncJob), Object(Illuminate\\Events\\CallQueuedListener))
#32 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Queue\\Jobs\\Job.php(98): Illuminate\\Queue\\CallQueuedHandler->call(Object(Illuminate\\Queue\\Jobs\\SyncJob), Array)
#33 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Queue\\SyncQueue.php(43): Illuminate\\Queue\\Jobs\\Job->fire()
#34 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Queue\\Queue.php(57): Illuminate\\Queue\\SyncQueue->push(Object(Illuminate\\Events\\CallQueuedListener), '', NULL)
#35 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Events\\Dispatcher.php(591): Illuminate\\Queue\\Queue->pushOn(NULL, Object(Illuminate\\Events\\CallQueuedListener))
#36 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Events\\Dispatcher.php(515): Illuminate\\Events\\Dispatcher->queueHandler('Statamic\\\\Listen...', 'handleSaved', Array)
#37 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Events\\Dispatcher.php(441): Illuminate\\Events\\Dispatcher->Illuminate\\Events\\{closure}(Object(Statamic\\Events\\AssetSaved))
#38 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Events\\Dispatcher.php(249): Illuminate\\Events\\Dispatcher->Illuminate\\Events\\{closure}('Statamic\\\\Events...', Array)
#39 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\helpers.php(451): Illuminate\\Events\\Dispatcher->dispatch('Statamic\\\\Events...')
#40 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Events\\Dispatchable.php(14): event(Object(Statamic\\Events\\AssetSaved))
#41 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Assets\\Asset.php(587): Statamic\\Events\\Event::dispatch(Object(Statamic\\Assets\\Asset))
#42 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Assets\\Asset.php(693): Statamic\\Assets\\Asset->save()
#43 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Collections\\HigherOrderCollectionProxy.php(60): Statamic\\Assets\\Asset->move('patrick-kavanag...')
#44 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Collections\\Traits\\EnumeratesValues.php(236): Illuminate\\Support\\HigherOrderCollectionProxy->Illuminate\\Support\\{closure}(Object(Statamic\\Assets\\Asset), 0)
#45 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Collections\\HigherOrderCollectionProxy.php(61): Illuminate\\Support\\Collection->each(Object(Closure))
#46 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Actions\\MoveAsset.php(40): Illuminate\\Support\\HigherOrderCollectionProxy->__call('move', Array)
#47 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Http\\Controllers\\CP\\ActionController.php(38): Statamic\\Actions\\MoveAsset->run(Object(Illuminate\\Support\\Collection), Array)
#48 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Controller.php(54): Statamic\\Http\\Controllers\\CP\\ActionController->run(Object(Illuminate\\Http\\Request))
#49 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\ControllerDispatcher.php(43): Illuminate\\Routing\\Controller->callAction('run', Array)
#50 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Route.php(260): Illuminate\\Routing\\ControllerDispatcher->dispatch(Object(Illuminate\\Routing\\Route), Object(Statamic\\Http\\Controllers\\CP\\Assets\\ActionController), 'run')
#51 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Route.php(205): Illuminate\\Routing\\Route->runController()
#52 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Router.php(727): Illuminate\\Routing\\Route->run()
#53 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(141): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#54 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Http\\Middleware\\DeleteTemporaryFileUploads.php(18): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#55 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Statamic\\Http\\Middleware\\DeleteTemporaryFileUploads->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#56 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Http\\Middleware\\CP\\CountUsers.php(18): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#57 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Statamic\\Http\\Middleware\\CP\\CountUsers->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#58 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Http\\Middleware\\CP\\Localize.php(19): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#59 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Statamic\\Http\\Middleware\\CP\\Localize->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#60 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Http\\Middleware\\CP\\Authorize.php(25): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#61 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Statamic\\Http\\Middleware\\CP\\Authorize->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#62 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Http\\Middleware\\CP\\AddToasts.php(22): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#63 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Statamic\\Http\\Middleware\\CP\\AddToasts->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#64 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Http\\Middleware\\CP\\AuthGuard.php(14): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#65 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Statamic\\Http\\Middleware\\CP\\AuthGuard->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#66 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Http\\Middleware\\CP\\ContactOutpost.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#67 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Statamic\\Http\\Middleware\\CP\\ContactOutpost->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#68 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Middleware\\SubstituteBindings.php(50): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#69 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Illuminate\\Routing\\Middleware\\SubstituteBindings->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#70 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken.php(78): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#71 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#72 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\View\\Middleware\\ShareErrorsFromSession.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#73 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#74 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Session\\Middleware\\StartSession.php(121): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#75 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Session\\Middleware\\StartSession.php(64): Illuminate\\Session\\Middleware\\StartSession->handleStatefulRequest(Object(Illuminate\\Http\\Request), Object(Illuminate\\Session\\Store), Object(Closure))
#76 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Illuminate\\Session\\Middleware\\StartSession->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#77 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#78 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#79 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Cookie\\Middleware\\EncryptCookies.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#80 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Illuminate\\Cookie\\Middleware\\EncryptCookies->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#81 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Http\\Middleware\\SwapExceptionHandler.php(19): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#82 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Statamic\\Http\\Middleware\\SwapExceptionHandler->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#83 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#84 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Router.php(728): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#85 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Router.php(705): Illuminate\\Routing\\Router->runRouteWithinStack(Object(Illuminate\\Routing\\Route), Object(Illuminate\\Http\\Request))
#86 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Router.php(669): Illuminate\\Routing\\Router->runRoute(Object(Illuminate\\Http\\Request), Object(Illuminate\\Routing\\Route))
#87 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Routing\\Router.php(658): Illuminate\\Routing\\Router->dispatchToRoute(Object(Illuminate\\Http\\Request))
#88 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Kernel.php(190): Illuminate\\Routing\\Router->dispatch(Object(Illuminate\\Http\\Request))
#89 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(141): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}(Object(Illuminate\\Http\\Request))
#90 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Http\\Middleware\\DisableFloc.php(18): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#91 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Statamic\\Http\\Middleware\\DisableFloc->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#92 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Http\\Middleware\\CheckMultisite.php(14): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#93 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Statamic\\Http\\Middleware\\CheckMultisite->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#94 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Http\\Middleware\\CheckComposerJsonScripts.php(14): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#95 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Statamic\\Http\\Middleware\\CheckComposerJsonScripts->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#96 C:\\webdev\\mysite\\vendor\\statamic\\cms\\src\\Http\\Middleware\\PoweredByHeader.php(19): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#97 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Statamic\\Http\\Middleware\\PoweredByHeader->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#98 C:\\webdev\\mysite\\vendor\\livewire\\livewire\\src\\DisableBrowserCache.php(19): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#99 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Livewire\\DisableBrowserCache->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#100 C:\\webdev\\mysite\\vendor\\barryvdh\\laravel-debugbar\\src\\Middleware\\InjectDebugbar.php(59): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#101 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Barryvdh\\Debugbar\\Middleware\\InjectDebugbar->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#102 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#103 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull.php(31): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#104 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#105 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#106 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\TrimStrings.php(40): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#107 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#108 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#109 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#110 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#111 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#112 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Http\\Middleware\\HandleCors.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#113 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Illuminate\\Http\\Middleware\\HandleCors->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#114 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Http\\Middleware\\TrustProxies.php(39): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#115 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(180): Illuminate\\Http\\Middleware\\TrustProxies->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#116 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Pipeline\\Pipeline.php(116): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#117 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Kernel.php(165): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#118 C:\\webdev\\mysite\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Http\\Kernel.php(134): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter(Object(Illuminate\\Http\\Request))
#119 C:\\webdev\\mysite\\public\\index.php(59): Illuminate\\Foundation\\Http\\Kernel->handle(Object(Illuminate\\Http\\Request))
#120 {main}
"} 
```

### Environment

```yaml
Version: 3.3.58 PRO
Laravel Version: 9.41.0
PHP Version: 8.1.12
```

### Installation

Fresh statamic/statamic site via CLI

**Comment by jasonvarga** [comment]
What addons do you have installed?

And do you have any custom event listeners? What's in your app/Providers/EventServiceProvider.php?

**Comment by stuartcusackie** [comment]
I have explained the structure below but I may have found a better clue. This seems to only happen on one particular file so far. The problematic file is called 'poet-patrick-kavanagh-in-field.jpg'. I have uploaded another file '001_text.jpg' and it moved without any warnings. I'm running more tests now to try and find more clues....

---------------

This project uses a custom routing system and event listeners to clear the static cache. There is only one custom event listener in EventServiceProvider.php:

```
EntrySaved::class => [
  ForceEntryInvalidation::class
]
```

I can share the contents of the above class if you want but I don't see why this would be fired when moving an asset - It's only registered on the EntrySaved event.

As for addons, I have a few:
```
justbetter/statamic-image-optimize: 1.1.3
spatie/statamic-responsive-images: 2.14.4
stillat/relationships: 1.2.1
stuartcusackie/statamic-cache-requester: 1.0.1
stuartcusackie/statamic-seo: 0.3.1
```
